### PR TITLE
fix: Added custom removal logic for databricks module & moved tenant-specific object IDs

### LINKED ISSUE #1581: [AVM Question/Feedback]: Databricks Connector not correctly removed
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Description

The `databricks/workspace` module creates an additional `-managed` resource group that is not removed by the CI. This can cause issues upon re-run if the location selector choses a different location to deploy to:

Example:
image

This means we need to investigate why this is the case, if there is a property we can set (similar to how you can specify on the VM that it should remove its NIC & PIP) or whether we can add something to the removal logic (e.g. adding it to the order in which resources are removed - to before the resource group is removed). 
```[tasklist]
### Tasks
```

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

- Added custom removal order item for the Databricks module to the CI.
- Moved Tenant-specifc object IDs from `dependencies.bicep` to `main.test.bicep` to make it easier to troubleshoot if one doesn't know and tries to troubleshoot the deployment in their own tenant (cc: @clintgrove, @rahalan)

Closes #1581

## Pipeline Reference

| Pipeline |
| -------- |
|    [avm.res.databricks.workspace](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.databricks.workspace.yml)      |
| [avm.res.container-instance.container-group](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.container-instance.container-group.yml) |

## Type of Change

- [x] Update to CI Environment or utilities (Non-module affecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

### CODE REVIEW

File: avm/utilities/pipelines/e2eValidation/resourceRemoval/helper/Get-DeploymentTargetResourceList.ps1
Comment (eriqua) [inline avm/utilities/pipelines/e2eValidation/resourceRemoval/helper/Get-DeploymentTargetResourceList.ps1:1]: This will be covered by another PR

File: avm/utilities/pipelines/e2eValidation/resourceRemoval/helper/Invoke-ResourcePostRemoval.ps1
Comment (eriqua) [inline avm/utilities/pipelines/e2eValidation/resourceRemoval/helper/Invoke-ResourcePostRemoval.ps1:1]: Any chance we can integrate this with the container group one so that we minimize the number of reruns?

File: avm/res/databricks/workspace/tests/e2e/waf-aligned/dependencies.bicep [hunk 1]
```diff
@@ -94,7 +94,8 @@ resource keyPermissions 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
   name: guid('msi-${keyVault::key.id}-${location}-${managedIdentity.id}-Key-Key-Vault-Crypto-User-RoleAssignment')
   scope: keyVault::key
   properties: {
-    principalId: '711330f9-cfad-4b10-a462-d82faa92027d' // AzureDatabricks Enterprise Application Object Id (Note: this is tenant specific)
+    // principalId: '711330f9-cfad-4b10-a462-d82faa92027d' // AzureDatabricks Enterprise Application Object Id (Note: this is tenant specific)
+    principalId: '5167ea7a-355a-466f-ae8b-8ea60f718b35' // AzureDatabricks Enterprise Application Object Id (Note: this is tenant specific)
```
Comment (AlexanderSehr) [inline avm/res/databricks/workspace/tests/e2e/waf-aligned/dependencies.bicep:98]: I wonder @clintgrove & @eriqua if we should move the value into the `main.test.bicep` - just to make it more obvious.

For example, we could add a parameter `param databricksApplicationObjectId string` to the dependencies.bicep and assign it a value in the `main.test.bicep` where all the other values are configured.

It doesn't matter 'per se' as it works as it is, but it took me 5 retries in my fork until I figured 'Maybe I missing a tenant-specific value in a test file and that's why it fails' 😄

File: avm/utilities/pipelines/e2eValidation/resourceRemoval/helper/Invoke-ResourcePostRemoval.ps1
Comment (AlexanderSehr) [reply avm/utilities/pipelines/e2eValidation/resourceRemoval/helper/Invoke-ResourcePostRemoval.ps1:1]: Hey @eriqua, I actually removed the code and started the test again. 

If we want to integrate this with the other one regardless, we can. Just say the word and I'll hold you to reviewing & approving the PR ;)

File: avm/res/databricks/workspace/tests/e2e/waf-aligned/dependencies.bicep (hunk 1 above)
Comment (clintgrove) [reply avm/res/databricks/workspace/tests/e2e/waf-aligned/dependencies.bicep:98]: HI @AlexanderSehr yes  I have noted this as well. But my thoughts are to NOT add a parameter for this, as if this was a scenario where a customer would use it, they would not have a parameter, as there is only one AppID for databricks in entire tenant, so making a parameter is pointless. I suggest we leave it like it is

**Comment by clintgrove** [comment]
Can you explain how you dealt with the ManagementResourceId for the "default" as it does not have an explicity assigned RG in our code?

File: avm/res/databricks/workspace/tests/e2e/max/dependencies.bicep [hunk 2]
```diff
@@ -94,7 +94,8 @@ resource keyPermissions 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
   name: guid('msi-${keyVault::key.id}-${location}-${managedIdentity.id}-Key-Key-Vault-Crypto-User-RoleAssignment')
   scope: keyVault::key
   properties: {
-    principalId: '711330f9-cfad-4b10-a462-d82faa92027d' // AzureDatabricks Enterprise Application Object Id (Note: this is tenant specific)
+    // principalId: '711330f9-cfad-4b10-a462-d82faa92027d' // AzureDatabricks Enterprise Application Object Id (Note: this is tenant specific)
+    principalId: '5167ea7a-355a-466f-ae8b-8ea60f718b35' // AzureDatabricks Enterprise Application Object Id (Note: this is tenant specific)
```
Comment (clintgrove) [inline avm/res/databricks/workspace/tests/e2e/max/dependencies.bicep:98]: I'd take this out, leave it how it was. This 5167ea is probably your tenants app id for databricks

Comment (AlexanderSehr) [reply avm/res/databricks/workspace/tests/e2e/max/dependencies.bicep:98]: It is. It's a draft PR and I need it for testing 😉

File: avm/res/databricks/workspace/tests/e2e/waf-aligned/dependencies.bicep (hunk 1 above)
Comment (AlexanderSehr) [reply avm/res/databricks/workspace/tests/e2e/waf-aligned/dependencies.bicep]: Agreed offline

**Review by rahalan — DISMISSED** [review]

File: avm/utilities/pipelines/e2eValidation/resourceRemoval/helper/Invoke-ResourcePostRemoval.ps1
Comment (eriqua) [reply avm/utilities/pipelines/e2eValidation/resourceRemoval/helper/Invoke-ResourcePostRemoval.ps1:1]: I mean #2180 as it's updating the same removal order list

Comment (AlexanderSehr) [reply avm/utilities/pipelines/e2eValidation/resourceRemoval/helper/Invoke-ResourcePostRemoval.ps1:1]: ok, that one is blocked by the AVM subscription currently (hence the draft). Will try to test it somewhere else. But if it remains blocked, I'd recommend to not block this PR over it. Will keep you posted

Comment (AlexanderSehr) [reply avm/utilities/pipelines/e2eValidation/resourceRemoval/helper/Invoke-ResourcePostRemoval.ps1:1]: Worked in my fork. Merged it into this PR @eriqua

**Review by eriqua — APPROVED** [review]
🆗
